### ISSUE TITLE
Client keeps syncing after calling logout(true) (or stopClient())

### ISSUE THREAD

**Issue by finsterwalder** [matrix-org/matrix-js-sdk#3678]
I initialized a client with a wrong token.
I attached a listener for HttpApiEvent.SessionLoggedOut, so it got triggered and I called mxclient.logout(true).
I see in the logs:
```
stopping MatrixClient
SyncApi.stop
```
But the client keeps syncing and my listener for ClientEvent.Sync is called regularly with state ERROR.

What is going wrong? How can I ensure the client stops, no matter if it's connected or not?

**Comment by finsterwalder** [comment]
I noticed, that FetchHttpApi logs an error:
```
FetchHttpApi: --> GET http://<someurl>:8008/_matrix/client/versions
FetchHttpApi: <-- GET http://<someurl>:8008/_matrix/client/versions [0ms AbortError: Failed to execute 'fetch' on 'Window': The user aborted a request.]
```
So the logout has the effect, that no requests are actually sent, but the sync is still triggered regularly...

**Comment by vitonsky** [comment]
I have the same problem

**Comment by finsterwalder** [comment]
Update your SDK.
I solved this issue with this Pull Request 
https://github.com/matrix-org/matrix-js-sdk/pull/3720

**Comment by vitonsky** [comment]
> Please update your SDK. I solved this issue with this Pull Request #3720

It means we have regression then. I just tried it and i decided to not use current package, because i can't stop it properly.
This is why i decided to use https://github.com/turt2live/matrix-bot-sdk package instead. They are no have docs too, so it is difficult to use and still does not support encrypted rooms fine, but at least i can stop their client.

**Comment by finsterwalder** [comment]
I don't think we have a regression of the same issue. I added tests to verify it. 
Your scenario must be different somehow.
Do you have a working connection that you want to stop? That works without problems and always has. My issue only happened when the initial connection had a problem and went into a retry. And during this retry a stop had no effect, which I fixed. 
Maybe you can describe exactly what is happening?
